engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkEngineRuntimeContextImpl.scala

```diff
@@ -4,13 +4,22 @@ import org.apache.flink.api.common.functions.RuntimeContext
 import pl.touk.nussknacker.engine.api.JobData
 import pl.touk.nussknacker.engine.api.runtimecontext.{ContextIdGenerator, IncContextIdGenerator}
 import pl.touk.nussknacker.engine.flink.api.FlinkEngineRuntimeContext
-import pl.touk.nussknacker.engine.util.metrics.MetricsProviderForScenario
+import pl.touk.nussknacker.engine.util.metrics.{MetricsProviderForScenario, NoOpMetricsProviderForScenario}
 
 case class FlinkEngineRuntimeContextImpl(jobData: JobData, runtimeContext: RuntimeContext)
     extends FlinkEngineRuntimeContext {
   override val metricsProvider: MetricsProviderForScenario = new FlinkMetricsProviderForScenario(runtimeContext)
 
   override def contextIdGenerator(nodeId: String): ContextIdGenerator =
     new IncContextIdGenerator(jobData.metaData.id + "-" + nodeId + "-" + runtimeContext.getIndexOfThisSubtask)
+}
+
+case class FlinkTestEngineRuntimeContextImpl(jobData: JobData, runtimeContext: RuntimeContext)
```

Comment:
DRY. It looks like the original author of `FlinkEngineRuntimeContextImpl` cut the corners and instead of having logic of creation of `MetricsProviderForScenario`, moved this logic into the body of the class. We shouldn't follow this approach but clean up this.

Reply:
I'm not sure if that's what you meant, if not then I'll need some help with it. I've tried to base the `MetricsProviderForScenario` on the `ComponentUseCase` but there is one place where it doesn't work, as a serialization error is thrown (see comment below).
Apart from this problem, everywhere else one method is used which looks the same for test and normal use, so there is less code duplication. But again, I'm not sure if it's enough